Emrald-UI/src/components/forms/VariableForm/VariableFormContext.tsx

```diff
@@ -0,0 +1,262 @@
+  const [sim3DId, setSim3DId] = useState<string>();
+  const [resetOnRuns, setResetOnRuns] = useState<boolean>();
+  const [docType, setDocType] = useState<string>();
+  const [docPath, setDocPath] = useState<string>();
+  const [docLink, setDocLink] = useState<string>();
+  const [pathMustExist, setPathMustExist] = useState<boolean | undefined>();
+
+  const { updateVariable, createVariable } = useVariableContext();
+
+  const sortNewStates = (newStateItems: AccrualStateItem[]) => {
+    return newStateItems.sort((a, b) => {
+      if (a.stateName && !b.stateName) {
+        return 1;
+      }
+      if (!a.stateName && b.stateName) {
+        return -1;
+      }
+      return 0;
+    });
+  };
+  const InitializeForm = (variableData?: Variable | undefined) => {
+    if (variableData?.name) {
+      setName(variableData.name);
+      const prefix: string = variableData.name.split('_')[0] + '_';
+      setNamePrefix(prefix);
+    } else {
+      setNamePrefix('Int_');
+    }
+    setDesc(variableData?.desc || '');
+    setType(variableData?.type || 'int');
+    setVarScope(variableData?.varScope || 'gtGlobal');
+    variableData?.value && setValue(String(variableData.value));
+    variableData?.sim3DId && setSim3DId(variableData.sim3DId);
+    variableData?.resetOnRuns && setResetOnRuns(variableData.resetOnRuns);
+    variableData?.docType && setDocType(variableData.docType);
+    variableData?.docPath && setDocPath(variableData.docPath);
+    variableData?.docLink && setDocLink(variableData.docLink);
+    variableData?.pathMustExist && setPathMustExist(variableData.pathMustExist);
+    variableData?.accrualStatesData &&
+      setAccrualStatesData(variableData.accrualStatesData);
+  };
+
+    // Maps 'type' values to their corresponding prefixes.
+    const PREFIXES: Record<string, string> = {
+      string: 'Str_',
+      double: 'Dbl_',
+      bool: 'Bool_',
+      default: 'Int_',
+    };
+  
+    const handleTypeChange = (newType: VariableType) => {
+      // Determine the prefix based on the 'type' value, or use 'default' if not found.
+      const updatedPrefix: string = PREFIXES[newType] || PREFIXES.default;
+      setNamePrefix(updatedPrefix);
+  
+      // Extract the part of the name after the prefix.
+      const nameWithoutPrefix: string = name ? name.split('_')[1] : '';
+  
+      // Set the 'name' state variable with the updated prefix and the extracted part.
+      setName(`${updatedPrefix}${nameWithoutPrefix}`);
+  
+      if (newType === 'bool') setValue('');
+    };
+  
+    const handleNameChange = (updatedName: string) => {
+      // Check if the updated name already contains the prefix
+      if (namePrefix) {
+        const hasPrefix = updatedName.startsWith(namePrefix);
+  
+        // Set the name with the appropriate prefix
+        setName(hasPrefix ? updatedName : `${namePrefix}${updatedName}`);
+      }
+    };
+  
+    const handleSave = (variableData?: Variable) => {
+      const newVariable = {
```

Comment:
I think it would look nicer if we could condense these into a signal. I need to do this for other forms too but I have a pretty good example in the actionFormContext
```suggestion
      variable.value = { // You would need to create this signal towards the top of the file.
        ...variable.value,
        id: variableData?.id || uuidv4(),
        type,
        name,
        desc,
        varScope,
        sim3DId,
        docType: docType as DocVarType,
        docPath,
        docLink,
        pathMustExist,
        value,
        accrualStatesData,
        resetOnRuns,
    }

    variableData
      ? updateVariable(varialbe.value)
      : createVariable(variable.value);
    handleClose();
```

Reply:
The signal should be added, let me know if I did it correctly. It seems to be working fine, but I'm not the signal expert